architecture.md

```diff
@@ -0,0 +1,565 @@
+# Architecture Guide
+
+
+This section contains documentation on the architecture of the Elys chain, including the current design and components of the system.
+
+<details>
+<summary>Click to expand/collapse</summary>
+
+## Boilerplate Generation
+
+The boilerplate was generated using `ignite CLI`, which provides a convenient way to generate new chains, modules, messages, and more. The initial modules that are part of the repository include `AssetProfile` and `LiquidityProvider`, both of which were generated using the `ignite CLI`.
```

Comment:
Do we need to keep this Boilerplate generation section?

Reply:
@jelysn yes I would still keep that as we are using `ignite-cli` for boilerplate generation and for development environment using the `config.yml` file, that’s kind of an intro for community devs willing to test out the code.